pkg/networkservice/common/connect/server_test.go

```diff
@@ -486,25 +509,30 @@ func TestConnectServer_DialTimeout(t *testing.T) {
 
 	// 1. Create connectServer
 
+	chainCtx, chainCancel := context.WithCancel(context.Background())
+	defer chainCancel()
+
 	s := next.NewNetworkServiceServer(
 		metadata.NewServer(),
-		connect.NewServer(context.Background(),
-			func(_ context.Context, cc grpc.ClientConnInterface) networkservice.NetworkServiceClient {
-				return networkservice.NewNetworkServiceClient(cc)
-			},
-			connect.WithDialTimeout(100*time.Millisecond),
-			connect.WithDialOptions(grpc.WithInsecure()),
+		connect.NewServer(
+			next.NewNetworkServiceClient(
+				dial.NewClient(chainCtx,
+					dial.WithDialTimeout(100*time.Millisecond),
+					dial.WithDialOptions(grpc.WithInsecure(), grpc.WithBlock()),
+				),
+				connect.NewClient(),
+			),
 		),
 	)
 
 	// 2. Setup fake A
 
 	listener, err := net.Listen("tcp", "127.0.0.1:")
 	require.NoError(t, err)
-	defer func() { _ = listener.Close() }()
 
 	urlA, err := url.Parse("tcp://" + listener.Addr().String())
 	require.NoError(t, err)
+	_ = listener.Close()
```

Comment:
Since we are actually timing out the Dialer at this point, we actually need to *not* have a listener to talk to.